src/main/java/com/direwolf20/buildinggadgets/common/building/placement/SingleTypeProvider.java

```diff
@@ -0,0 +1,44 @@
+package com.direwolf20.buildinggadgets.common.building.placement;
+
+import com.direwolf20.buildinggadgets.common.building.IBlockProvider;
+import com.direwolf20.buildinggadgets.common.building.TranslationWrapper;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTUtil;
+import net.minecraft.util.math.BlockPos;
+
+public class SingleTypeProvider implements IBlockProvider {
```

Comment:
Please add some description of what it does. This especially important with this class.